rotkehlchen/inquirer.py

```diff
@@ -369,10 +400,25 @@ def find_usd_price(
             except RemoteError:
                 pass  # continue, a price can be found by one of the oracles (CC for example)
 
+        # Try and check if it is an ethereum token with specified protocol or underlying tokens
+        is_known_protocol = False
+        underlying_tokens = None
+        try:
+            token = EthereumToken.from_asset(asset)
+            if token is not None:
+                if token.protocol is not None:
+                    is_known_protocol = token.protocol in KnownProtocolsAssets
+                underlying_tokens = GlobalDBHandler().get_ethereum_token(  # type: ignore
+                    token.ethereum_address,
+                ).underlying_tokens
+        except UnknownAsset:
+            pass
+
+        # Check if it is a special token
         if asset in instance.special_tokens:
             ethereum = instance._ethereum
             assert ethereum, 'Inquirer should never be called before the injection of ethereum'
-            token = EthereumToken.from_asset(asset)
+            # If we are dealing with and eth token, token should be defined already
```

Comment:
I don't understand the comment.

> If we are dealing with and eth token

I don't think you need this at all. Right below the text of the assertion says the same thing, no?

Reply:
Yes below it says nearly the same. What I wanted to say is that if the code enters this branch, the variable defined at https://github.com/rotki/rotki/pull/2859/files#diff-93cbb20f918b107a6f137a204f702d1257fc49e21bdd62e6d557732cfb4b502eR407 should be correctly initialize

Reply:
yes this is what the assertion is for. Please remove the comment.

> If we are dealing with and eth token, token should be defined already